### ISSUE TITLE
What's the difference between validation data and eval data?

### ISSUE THREAD

**Issue by zzxiang** [espnet/espnet#3875]
During training, the data is divided into three groups: training data, validation (also known as dev) data and eval data.

The training data is used to train the neural network.

The validation data is used to compute validation loss applying the trained network to let us prevent over-fitting.

But what about eval data?

**Comment by kan-bayashi** [comment]
The evaluation data (= test data) is a set used to provide an unbiased evaluation of the model.
The evaluation data must be unknown for the model during the training to calculate the performance without any cheating.

**Comment by zzxiang** [comment]
I thought the validation data is also unknown for the model. It is used only to compute some indicators such as loss values, but has no effect on the training of network. So doesn't the validation data provide the same evaluation of the model?

**Comment by kan-bayashi** [comment]
Validation data is not used for updating the parameters, but it is often used to decide the best parameters (e.g., early stopping).
In that case, the model is decided to maximize the performance of the validation data, and therefore, the evaluation result for the validation provide a kind of biased results.

**Comment by zzxiang** [comment]
I see. So a model with good performance on validation data does not necessarily mean it has good generalization ability (i.e. such as in TTS, can synthesize good speech for any text input).

Thanks for kindly explanation. Seems that it's a basic knowledge of machine learning.

I wonder where can I check the evaluation result? Seems that tensorboard only shows the train and validation results.

**Comment by kan-bayashi** [comment]
> So a model with good performance on validation data does not necessarily mean it has good generalization ability.

Right.
Most of the classification task models monitors the validation score to decide the best parameters and it leads the reasonable score for evaluation data in the most of cases.  
However, in the case of TTS (generative models), the validation score monitoring is not the best way to decide. Even if the model show the overfitting tendency for the validation score, the models often generate better speech than the validation best model.
The monitoring way of TTS models is one of the research items.

> I wonder where I can check the evaluation result? Seems that tensorboard only shows the train and validation results.

In the case of ASR, you can find the result.txt representing the WER / TER / CER for the evaluation sets.
In the case of TTS, the objective results are not calculated in the recipe but you can calculate the following measures manually.
https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/tts1/README.md#evaluation
In my experience, the ASR score for the generated speech is good measure to check the quality.
But MCD and F0 are not so good measure for E2E-TTS models, which are just of a  kind of the reference record.

**Comment by zzxiang** [comment]
Thank you very much for the explanation!
